code/modules/experisci/experiment/handlers/experiment_handler.dm

```diff
@@ -0,0 +1,393 @@
+		if(EXPERIMENT_CONFIG_UI)
+			RegisterSignal(parent, COMSIG_UI_ACT, .proc/ui_handle_experiment)
+
+	// Auto connect to the first visible techweb (useful for always active handlers)
+	// Note this won't work at the moment for non-machines that have been included
+	// on the map as the servers aren't initialized when the non-machines are initializing
+	if (!(config_flags & EXPERIMENT_CONFIG_NO_AUTOCONNECT))
+		var/list/found_servers = get_available_servers(parent)
+		var/obj/machinery/rnd/server/s = found_servers.len ? found_servers[1] : null
+		if (s)
+			link_techweb(s.stored_research)
+
+	GLOB.experiment_handlers += src
+
+/datum/component/experiment_handler/Destroy(force, silent)
+	. = ..()
+	GLOB.experiment_handlers -= src
+
+/**
+ * Hooks on attack to try and run an experiment (When using a handheld handler)
+ */
+/datum/component/experiment_handler/proc/try_run_handheld_experiment(datum/source, atom/target, mob/user, params)
+	SIGNAL_HANDLER
+	if (!should_run_handheld_experiment(source, target, user, params))
+		return
+	INVOKE_ASYNC(src, .proc/try_run_handheld_experiment_async, source, target, user, params)
+	return COMPONENT_CANCEL_ATTACK_CHAIN
+
+/**
+ * Provides feedback when an item isn't related to an experiment, and has fully passed the attack chain
+ */
+/datum/component/experiment_handler/proc/ignored_handheld_experiment_attempt(datum/source, atom/target, mob/user, proximity_flag, params)
+	SIGNAL_HANDLER
+	if (!proximity_flag || (selected_experiment == null && !(config_flags & EXPERIMENT_CONFIG_ALWAYS_ACTIVE)))
+		return
+	playsound(user, 'sound/machines/buzz-sigh.ogg', 25)
+	to_chat(user, "<span>\the [target.name] is not related to your currently selected experiment.</span>")
+
+/**
+ * Checks that an experiment can be run using the provided target, used for preventing the cancellation of the attack chain inappropriately
+ */
+/datum/component/experiment_handler/proc/should_run_handheld_experiment(datum/source, atom/target, mob/user, params)
+	// Check that there is actually an experiment selected
+	if (selected_experiment == null && !(config_flags & EXPERIMENT_CONFIG_ALWAYS_ACTIVE))
+		return
+
+	// Determine if this experiment is actionable with this target
+	var/list/arguments = list(src)
+	arguments = args.len > 1 ? arguments + args.Copy(2) : arguments
+	if (config_flags & EXPERIMENT_CONFIG_ALWAYS_ACTIVE)
+		for (var/datum/experiment/e in linked_web.available_experiments)
+			if (e.actionable(arglist(arguments)))
+				return TRUE
+	else
+		. = selected_experiment.actionable(arglist(arguments))
+
+/**
+ * This proc exists because Jared Fogle really likes async
+ */
+/datum/component/experiment_handler/proc/try_run_handheld_experiment_async(datum/source, atom/target, mob/user, params)
+	if (selected_experiment == null && !(config_flags & EXPERIMENT_CONFIG_ALWAYS_ACTIVE))
+		to_chat(user, "<span>You do not have an experiment selected!.</span>")
```

Comment:
Here and everywhere else you do it, these need to have classes. `notice`, `warning`, I don't care, but not just <span>.